catalogue/__init__.py

```diff
@@ -132,11 +142,17 @@ def get_entry_point(self, name: str, default: Optional[Any] = None) -> Any:
         default (Any): The default value to return.
         RETURNS (Any): The loaded entry point or the default value.
         """
-        for entry_point in AVAILABLE_ENTRY_POINTS.get(self.entry_point_namespace, []):
+        for entry_point in self._get_entry_points():
             if entry_point.name == name:
                 return entry_point.load()
         return default
 
+    def _get_entry_points(self) -> List[importlib_metadata.EntryPoint]:
+        if isinstance(AVAILABLE_ENTRY_POINTS, SelectableGroups):
+            return cast(AVAILABLE_ENTRY_POINTS, SelectableGroups).select(group=self.entry_point_namespace)
```

Comment:
The cast shouldn't be necessary after `isinstance`.
```suggestion
            return AVAILABLE_ENTRY_POINTS, SelectableGroups.select(group=self.entry_point_namespace)
```